vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/LinkManager.java

```diff
@@ -302,7 +304,79 @@ public IModuleLinkDisplay createModuleLink(String title, ISpace source, float po
         moduleLinkDisplayRepo.save((ModuleLinkDisplay) display);
         return display;
     }
+    
+    @Override
+    public IModuleLinkDisplay editModuleLink(String title, ISpace source, float positionX, float positionY,
+            int rotation, String linkedModuleId, String moduleLinkLabel, String linkId, String moduleLinkDisplayId)
+            throws SpaceDoesNotExistException {
+        
+        source = spaceManager.getSpace(source.getId());
+        if (source == null) {
+            throw new SpaceDoesNotExistException();
+        }
+        Optional<ModuleLink> linkOptional = moduleLinkRepo.findById(linkId);
+        IModuleLink link = linkOptional.get();
+        link.setName(title);
+        IModule module = new Module();
+        module.setId(linkedModuleId);
+        link.setModule(module);
```

Comment:
that seems odd, why don't you get the actual module it links to and set it here?